### ISSUE TITLE
iget gives stale replica  *with no flags*

### ISSUE THREAD

**Issue by bh9** [irods/irods#4309]
## Feature

- Please describe the feature you are requesting and your use case.

OR

### iRODS Version, OS and Version
4.1.12, Ubuntu Precise 12.04

### What did you try to do?
iget an object with 1 good replica and 1 "stale" replica

### Expected behavior
iget gives me the good replica

### Observed behavior (including steps to reproduce, if applicable)
iget gives me the *first* replica, regardless of whether it is good or not, here replica 0 is stale and replica 1 is good, but replica 0 is returned:
```
irods10kg1@irods-uk10k-g1:~$ ils -L $FILE
  jm23              0 root;random;uk10k-green;irods-data-GU1-2      8691168 2012-04-27.15:26   UK10K_ALS5085673.bam.bai
    845a3a5f15565d61b3c592bdf79b76d7    generic    /irods-sd-bc/replica/uk10k/UK10K_COHORT_ALSPAC/3/5/6/8/3568016b1f1775ec54feab677fcc8107/UK10K_ALS5085673.bam.bai.984604634
  jm23              1 root;random;uk10k-green;irods-data-GU1-2      8691168 2013-03-14.01:09 & UK10K_ALS5085673.bam.bai
    0cb157aa1bfc956179a17c6043f3472c    generic    /irods-sd-bc/uk10k/UK10K_COHORT_ALSPAC/3/5/6/8/3568016b1f1775ec54feab677fcc8107/UK10K_ALS5085673.bam.bai
irods10kg1@irods-uk10k-g1:~$ iget $FILE - |md5sum
845a3a5f15565d61b3c592bdf79b76d7  -
```

**Comment by bh9** [comment]
This is on our unreplicated zone, in case that makes a difference

**Comment by jasoncoposky** [comment]
It appears that both of those replicas are on the same leaf resource `irods-data-GU1-2` which is the real issue.  The reason you get the stale replica is simply because it is first in the list for that resource.   I would suggest you trim the stale replica and rebalance if this is in a replication hierarchy.  Multiple replicas should not be stored on a single leaf resource.

**Comment by bh9** [comment]
I found this while preparing to trim the stale replica. I was making a copy of each good replica before running the delete just in case something went wrong (nothing did in the end) with itrimming 1700 replicated files. Unfortunately we don't know how it got into that state, but the affected files were created years ago on an older version of iRODS (probably 3.3.1)

**Comment by trel** [comment]
Yes, I see the timestamp of 2012/2013.   I think this is a wontfix at this point - we have to trust the catalog for this operation.

If you're willing to reproduce... Can you confirm that an `iget -K` would have caught the bad replica on the way out?

**Comment by trel** [comment]
@bh9 please let us know if you get a chance to test that using the checksum would have caught the bad replica.
